### ISSUE TITLE
[FEATURE_REQUEST] Migrate Cohere v1/chat endpoint to v2/chat

### ISSUE THREAD

**Issue by cloak1505** [SillyTavern/SillyTavern#2916]
### Have you searched for similar requests?

Yes

### Describe the solution you'd like

No longer labelled as beta today.
https://docs.cohere.com/v2/reference/chat-v2

### Additional context

The documentation shows `p` range of .01-.99 but it doesn't error out on 0-1, at least on Python. I notice ST currently clamps 1 down to .99. I believe we should include 0-1 for consistency.
Temp is still 0-1 (not surprisingly; I'm just saying this since it doesn't say in docs).
`safety_mode` is the same deal, errors on models older than August refresh (we already know this).

**Last message still needs to be "user" role,** sadly or else it errors. But you can have any number of messages and their role in any order in the `messages` array, except enforcing last as user. This means "Squash system messages" option is usable, letting users decide whether the system prompts are combined or not.

### Priority

Low (Nice-to-have)

### Are you willing to test this on staging/unstable branch if this is implemented?

Yes

**Comment by cloak1505** [comment]
Hrm... I did prefer TC "Skip Example Dialogues Formatting" style which v1 had since it was put in `preamble`, ~~but I imagine "normal ST style" entails having turns.~~

(Edit 2: *DON'T* do below, RisuAI does this and it's jank. I end up having to use `{{#if {{getglobalvar::toggle_exampleVarName}} }} {{/if}}` macro as custom prompt (cut & paste examples so it's no longer in card) + Chat Bot > Others > Prompt Template > `exampleVarName=Mes Example: Char` in Custom Toggles field which will display a checkbox item labeled "Mes Example: Char" on left side of chat UI, and manually toggle per character.)
```
  messages=[
    {'role': 'system', 'content': '[Example Chat]'},
    {'role': 'user', 'content': '"Hi."'},
    {'role': 'assistant', 'content': '"Go away," Char grumbles.'},
    {'role': 'system', 'content': '[Start a new Chat]'}
  ]
```

Edit: Never mind, ignore that! I just remembered ST puts example dialogues in system messages, but with 'name' object field for some endpoints.
